local/bin/py/build/pull_rbac.py

```diff
@@ -31,18 +41,42 @@ def pull_rbac():
             print(f'\x1b[33mWARNING\x1b[0m: RBAC api request failure reason: {e}')
             pass
 
-    if json_result:
-        data = json_result.get('data', [])
+    if permissions_json and roles_json:
+        permissions_data = permissions_json.get('data', [])
+        # gets default roles (admin, standard, and read-only)
+        # ignores UI template roles ('Datadog Billing Admin Role' and 'Datadog Security Admin Role')
+        default_roles_data = roles_json.get('data', [])[0:3] 
+        dr_data_len = len(default_roles_data)
 
-        for permission in data:
+        for permission in permissions_data:
             group_name = permission['attributes']['group_name']
             permission_name = permission['attributes']['name']
+            permission_description = permission['attributes']['description']
+            permission_role_name = ''
+
+            for x in range(dr_data_len):
+                if permission_role_name:
+                    break
+                role = default_roles_data[(dr_data_len-1)-x] # get least permissive role starting from the 'Datadog Read Only Role' -> 'Datadog Admin Role'
+                role_name = role['attributes']['name']
+                role_permissions = role['relationships']['permissions']['data']
 
-            # Remove legacy logs permissions from dictionary before converting to JSON.  These legacy permissions are hard-coded in rbac-permissions-table partial until they can be deprecated.
-            if permission_name in ('logs_live_tail', 'logs_read_index_data'):
+                for role_permission in role_permissions:
+                    # Match permission id to a role. assign permission_role_name
+                    # assign permission_role_name - relies on order of Admin, Standard, and Read Only role objects in the `roles/templates` api.
+                    if role_permission['id'] == permission['id']:
+                        permission_role_name = role_name
+                        if permission_role_name:
+                            break
```

Comment:
Do we need this check, won't it always have a value as we are setting it the line above?

Reply:
good call! Updated